TypeSystem/Reflection/test.cc

```diff
@@ -509,4 +725,48 @@ TEST(Reflection, BaseTypeMatters) {
                                 Reflector().ReflectType<two::IdenticalCurrentStructWithDifferentBaseType>()).type_id));
 }
 
+namespace reflection_test {
+
+CURRENT_FORWARD_DECLARE_STRUCT(One);
+CURRENT_FORWARD_DECLARE_STRUCT(Two);
+
+CURRENT_STRUCT(One) { CURRENT_FIELD(two, std::vector<Two>); };
+
+CURRENT_STRUCT(Two) { CURRENT_FIELD(one, std::vector<One>); };
+
+}  // namespace reflection_test
+
+TEST(Reflection, OrderDoesNotMatter) {
+  using namespace reflection_test;
+  auto const one = current::reflection::CurrentTypeID<One>();
+  auto const two = current::reflection::CurrentTypeID<Two>();
+  EXPECT_EQ(9201293424144532153ull, static_cast<uint64_t>(one));
+  EXPECT_EQ(9207292435550686765ull, static_cast<uint64_t>(two));
+  // `Reflector` is thread local, so starting another thread is the easiest way to ensure a stateless run.
+  // Also note that the order of computations of TypeIDs is flipped in the two threads.
+  std::thread([one, two]() {
+    EXPECT_EQ(static_cast<uint64_t>(one), static_cast<uint64_t>(current::reflection::CurrentTypeID<One>()));
+    EXPECT_EQ(static_cast<uint64_t>(two), static_cast<uint64_t>(current::reflection::CurrentTypeID<Two>()));
+  }).join();
+  std::thread([one, two]() {
+    EXPECT_EQ(static_cast<uint64_t>(two), static_cast<uint64_t>(current::reflection::CurrentTypeID<Two>()));
+    EXPECT_EQ(static_cast<uint64_t>(one), static_cast<uint64_t>(current::reflection::CurrentTypeID<One>()));
+  }).join();
+}
+
+namespace reflection_test {
+
+CURRENT_STRUCT_T(RSRange) {
```

Comment:
It's essential to confirm the numerical value or the type ID for this `RandomShitMinMaxRange` remains what it is.